app/models/user.rb

```diff
@@ -57,6 +59,19 @@ class User < ActiveRecord::Base
   scope :current_students, -> { Role.where({ name: Role::STUDENT, semester: Semester.last })
     .includes(:user).map(&:user) }
 
+  def assignment_submission_pairs
+    pairs = []
+    assignments = Semester.current.assignments
+    assignments.each do |assignment|
+      submission = assignment_submissions.find { |x| x.assignment == assignment }
+      if submission.nil?
+        submission = AssignmentSubmission.new points: 0
+      end
```

Comment:
`if .... end ` can be simplified as `submission ||= AssignmentSubmission:new points:0`